### PR TITLE
Document product and engineering planning process

### PR DESCRIPTION
**Pull request by msennett22**
This creates a new section in the handbook, `product-engineering`, where we can document things that impact product and engineering equally (like planning), and then it adds documentation for our new planning process. The plan is to move the current `product` and `engineering` folders under this new folder, but that will not happen in this PR.

This PR includes and obsoletes the content in [this doc](https://docs.google.com/document/d/1vYMH3s7B8mP2jRx5_9qYOiVJLs2ZX6T1i9NFHM1xgGk/edit#).

### CODE REVIEW

**Comment by jasonyavorsky** [comment]
I love the idea of combining product & engineering under one heirarchy!

File: content/product-engineering/planning.md
Comment (jasonyavorsky) [inline content/product-engineering/planning.md]: There's some content at https://handbook.sourcegraph.com/product/product_management#okr-and-roadmap-tracking that could either:

- Be linked to from here
- Move here (and then link to here from there)

Comment (jasonyavorsky) [reply content/product-engineering/planning.md]: Not sure which is best.

File: content/product-engineering/planning.md
Comment (jasonyavorsky) [inline content/product-engineering/planning.md]: Around this point in time I think that updating https://handbook.sourcegraph.com/company/strategy should be an official tracked step. What do you think?

```suggestion
| Week 12 of current quarter<br>1 week until next quarter | End of quarter retrospective<li>VPP/VPE & Product/Eng Directors retro previous quarter<li>VPP/VPE and CTO team retro previous quarter & finalize OKR scorecard <li>Update [company strategy and sub-pages](../../company/strategy) |
```

File: content/product-engineering/planning.md
Comment (jasonyavorsky) [inline content/product-engineering/planning.md]: Adding a weekly section to go with monthly and quarterly so you can't miss it.

```suggestion
### Weekly team updates

These are due each Friday by 19:00 UTC.

- Update the [GitHub project tracker](#github-project-tracker)
- Update the [OKR slides](#okr-slides)

### Monthly team updates
```

**Comment by jasonyavorsky** [comment]
I made a few small commits to change things in the product handbook that were duplicated to be links to the new content.

File: content/product-engineering/index.md
Comment (jasonyavorsky) [inline content/product-engineering/index.md]:
```suggestion
The roles of [Product](../product/index.md) and [Engineering](../engineering/index.md) are to create value for our customers by delivering a high quality product that solves their problems.
```

**Comment by marybelzer** [comment]
Thoughts on moving the "Product" and Engineering" directories into the "Product-Engineering" directory? Having all three at the same level of the Handbook may make it a bit confusing to navigate and to know where certain things should live

**Comment by jasonyavorsky** [comment]
> 
> 
> Thoughts on moving the "Product" and Engineering" directories into the "Product-Engineering" directory? Having all three at the same level of the Handbook may make it a bit confusing to navigate and to know where certain things should live

@marybelzer that's listed as a todo in the description, but as a follow-up PR. Once this merges I'd be happy to take a shot at it, I've built up some tricks for handling large moves like that.

**Comment by nicksnyder** [comment]
Yes, we will do that, just not in this particular change so it is easier to review the new content in this change.

**Comment by msennett22** [comment]
I added the product/eng Q4 direction info and redacted customer/sensitive data. @nicksnyder please let me know if there is anything else we need to remove from this section due to sensitivity

File: content/product-engineering/planning.md
Comment (yinquanteo) [inline content/product-engineering/planning.md]: @nicksnyder is this something you want from all engineering teams?

I've gone back to my team to gather some initial feedback and overall group sentiment leans towards preferring Code Graph's current GitHub issues approach to providing team updates. Things people liked – process feels more lightweight, everything being in GitHub, better integration with GitHub (referencing issues/PRs). One person did point out they liked the email approach because it had a wider audience.

Regardless of the decision here, I'd love to minimize the need for duplicate updates.

@lguychard @scalabilitysolved @jjeffwarner @felixfbecker @chrispine tagging you here if wanted to comment directly

Comment (nicksnyder) [reply content/product-engineering/planning.md]: Yes the proposal documented in this PR is for each team to send a monthly email (instead of a monthly post to a GitHub issue for that team). The reason for email is that it is easier for stakeholders outside of engineering to subscribe to (only need to subscribe to a single mailing this rather than a bunch of issues), but we could try to automate that to bridge the gap if we need to. If we used one long lived issue per team, eventually they would get really long and hard to navigate.

The goal is for this not to be duplicative with weekly tactical updates on KRs. The audience is different, level of detail is different (due to monthly vs weekly cadence)

Definitely open to feedback here!

**Review by jasonyavorsky — APPROVED** [review]
Left a few ideas. Looks good!

File: content/product-engineering/planning.md
Comment (felixfbecker) [inline content/product-engineering/planning.md]: **issue**: The `<li>` (list item) tags in these tables are missing enclosing `<ul>` `</ul>` (unordered list) tags and closing `</li>` tags, which causes rendering issues, e.g. in Firefox:

image

```suggestion
| Week 6          | Finalize annual vision document <ul> <li>VPs update and finalize based on feedback</li> <li>Annual vision added to handbook & communicated on Slack</li> </ul> |
```

(needs to be done for all `<li>`s in this PR)

File: content/product-engineering/planning.md
Comment (msennett22) [reply content/product-engineering/planning.md]: @jyavorska In week 10 we have a step to add the OKRs to the handbook and communicate them on Slack. We can link to the pages as you suggested to ensure people know exactly where in the handbook it will be updated.

Comment (jasonyavorsky) [reply content/product-engineering/planning.md]: I missed that, thanks @msennett22 ! I'll add a suggestion up there instead.

File: content/product-engineering/planning.md
Comment (jasonyavorsky) [inline content/product-engineering/planning.md]:
```suggestion
| Week 10 of current quarter<br>2 weeks until next quarter | Finalize OKRs<li>VPP/VPE & Product/Eng Directors align on final OKRs during [VP team sync](team.md#vp-team-sync)<li>VPP/VPE get exec team buy in to finalize and approve OKRs<li>OKRs added to the handbook and [company strategy and sub-pages](../../company/strategy) updated & communicated on Slack |
```

**Review by jeanduplessis — APPROVED** [review]

**Review by christinaforney — APPROVED** [review]

File: content/product-engineering/direction.md
Comment (nicksnyder) [inline content/product-engineering/direction.md]:
```suggestion
Sourcegraph is a product that developers love, and once we land inside an organization usage grows organically! To set us up to grow our business in FY23 we need to repeatably drive more users to the point where they are using Sourcegraph on their private code with their team. To achieve this outcome, we are going to pursue three strategic objectives in FQ4.
```

File: content/company/strategy/index.md
Comment (jasonyavorsky) [inline content/company/strategy/index.md]: Propose renaming this to `strategy.md` to be consistent with what we're calling other strategy pages (cc @nicksnyder @serinadawn )

File: content/product-engineering/planning.md
Comment (msennett22) [inline content/product-engineering/planning.md]: Proposing to add a section about our philosophy/principles. It could potentially look something like this. 

##Planning philosophy and principles 
Our planning process defines how we meet and execute to create our annual and quarterly goals. Our philosophy is based on the following principles:

- Clear direction: by setting clear direction on the department level we are able to stay focused throughout the year and have appropriate trade-off discussions if necessary. 

- Coupling insights: our teams are the domain experts and their valuable insights coupled with the overall company strategy are critical to both enable and drive our strategic planning.

- Creating impact : to set ourselves up for success we want to understand and be able to measure the impact and value the work we commit to will have on customers, our internal teams, and the overall success of Sourcegraph.

File: content/company/strategy/index.md
Comment (nicksnyder) [reply content/company/strategy/index.md]: done

File: content/product-engineering/planning.md
Comment (jasonyavorsky) [inline content/product-engineering/planning.md]: Do we need both a manually written table of contents in addition to the automatically generated one to the right?

**Comment by jasonyavorsky** [comment]
This morning I made it so there isn't also a planning page under product management, instead it just links to the product-engineering planning page.

File: content/product-engineering/planning.md
Comment (jasonyavorsky) [reply content/product-engineering/planning.md]: @msennett22 FYI there is a philosophy section at `/product-engineering/planning#philosophy` (could be brand new, not sure) and it touches on some of these.

File: content/product-engineering/direction.md
Comment (jasonyavorsky) [inline content/product-engineering/direction.md]: Maybe rename this file to strategy.md? Not super important but it's more consistent.

File: content/product-engineering/planning.md
Comment (dan-mckean) [inline content/product-engineering/planning.md]:
```suggestion
- Everyone on our team has context and insights that are valuable inputs into the planning process.
```

File: content/product-engineering/planning.md
Comment (dan-mckean) [inline content/product-engineering/planning.md]: The start of this makes no sense and it's not clear what it was meant to say.

File: content/product-engineering/planning.md
Comment (dan-mckean) [inline content/product-engineering/planning.md]: Love this line ❤️

File: content/product-engineering/planning.md
Comment (dan-mckean) [inline content/product-engineering/planning.md]: I feel like this line—specifically the bit about collecting information from teams—belies a lot of work, and it would be worth somehow making this more explicit to ensure it happens properly, far enough in advance of the VP/Director meeting.

File: content/product-engineering/planning.md
Comment (dan-mckean) [inline content/product-engineering/planning.md]: What happens next? Do you share the learnings and give space for others outside of this small working group to suggest solutions?

File: content/product-engineering/team.md
Comment (dan-mckean) [inline content/product-engineering/team.md:24]: What is the CTO team? Worth being more explicit!

File: content/product-engineering/planning.md
Comment (jasonyavorsky) [inline content/product-engineering/planning.md]: Maybe add milestones as a fourth definition here because we're now tracking those as separate entities?

File: content/product-engineering/planning.md
Comment (felixfbecker) [inline content/product-engineering/planning.md]: Leftover TODO link here

@jyavorska  shouldn't the link checker have caught this? 🤔

Comment (jasonyavorsky) [reply content/product-engineering/planning.md]: @felixfbecker I removed the link checker I made because it didn't work properly. The only way to really make it work consistently is post-build and export.

File: content/product-engineering/planning.md
Comment (nicksnyder) [reply content/product-engineering/planning.md]: Yes, just added that based on feedback yesterday and this comment. Going to resolve this thread.

File: content/product-engineering/planning.md
Comment (nicksnyder) [reply content/product-engineering/planning.md]: I found this one helpful (and it is auto-generated by me via extension), but I will remove it. The index to the right in the handbook is missing two things IMO:
1. I think it should include H2s under H1s with indent.
1. Admittedly personal preference, but I think outlines should be on the left. On the right feels too hidden.

Issue filed here: https://github.com/sourcegraph/handbook/issues/901

Comment (jasonyavorsky) [reply content/product-engineering/planning.md]: I actually like them on the left too, and I don't like the way it hides h2s today. At the same time it does feel strange to see two copies of the table of contents side by side when you go to the page. Maybe there's an easy fix for the official table of contents.

File: content/product-engineering/planning.md
Comment (nicksnyder) [reply content/product-engineering/planning.md]: This whole section is still wip. Will be working on this more today.

**Comment by nicksnyder** [comment]
@christinaforney and I aligned on what we want out of the tracker today but I ran out of time to make any updates to this PR. I will keep working on this PR tomorrow and the rest of this week to get this merged ASAP. 

Here is the short version...

There are two kinds of line items: KRs and Roadmap items.
* KRs are the implementation-independent definition of success. Generally, these are based on metrics.
* Roadmap items are concrete things the team is delivering to achieve the definition of success. Generally, these are feature/product launches and each roadmap items has a "Target date" that the implementing team is responsible for setting and updating for accuracy.
  * For most rows, the target date just needs to be a best effort (conservative) estimate of when we will be done. Changes to the target date are ok, just log what changed and why in the linked issue. This helps me/Christina/CE/Sales/Marketing generally know when things are going to be available so we can plan ahead. The more accurate dates that you can provide, the quicker we will be able to get customers using what you have released on day 1 so you aren't waiting for feedback.
  * For a small number of rows that are our biggest launches and we want to do a big marketing effort around, we will need to have a more firm date that we can commit to (for embargo reasons) when we get closer to launch. We will ensure that the people responsible for these deliverables know what the expectations are.

Tracker tabs:
* Department KRs tab unchanged. It only contains top-level product/eng KRs. This is what we review on a weekly basis with product/eng directors and what we are reporting on to the rest of the business and exec team.
* Org KRs new tab. This has per-org level KRs. The expectation is that product/eng directors are reviewing these weekly with relevant teams. Me/Christina will deep-dive review these with product/eng directors monthly, and spot check as necessary or whenever something is escalated.
* Team deliverables new tab. This has top-level roadmap items per team so Christina and I know what the big picture of each team is working on. If when a team doesn't have a roadmap item, then we will include the KR as a placeholder, but once the team has identified at least one roadmap item, we will remove the KR from this view. Teams that have line of sight into multiple future deliverables are encouraged to include those, but don't artificially flesh out a roadmap that you don't have confidence in. 
* Per-org level tabs are owned by the directors for each org. Christina and I don't care about what is in those tabs or how they are used.

Christina and I also recorded a video running through the dashboard, but zoom is not done processing that yet....I'll post it in [#product-engineering-planning](https://sourcegraph.slack.com/archives/C02LHQW7RC0/p1637104051148100) when it is ready.


Would love feedback on this latest iteration of the proposal.

**Comment by jasonyavorsky** [comment]
> There are two kinds of line items: KRs and Roadmap items.

Does this mean the milestone issues we just added are going away?

**Comment by christinaforney** [comment]
> Does this mean the milestone issues we just added are going away?

No, we migrated all Milestones to be Roadmap items. The difference is the view level - if it shows up at Prod/Eng (Department) level vs. Org level.

**Comment by nicksnyder** [comment]
I am happy with where this is at so going to bias toward merging now and iterating on the content.

@jyavorska I removed some of the content that you added to the new planning page. I really want this page to be focused on the process itself and not the details of how to prioritize things.

File: content/product-engineering/planning.md
Comment (nicksnyder) [reply content/product-engineering/planning.md]: What work needs to happen is an implementation detail for the directors to figure out. This is just documenting the API.

**Comment by jasonyavorsky** [comment]
Sure @nicksnyder sorry those changes weren't more helpful. We can find a different way to disambiguate the two planning process pages in a later follow up.

File: content/product-engineering/planning.md
Comment (nicksnyder) [reply content/product-engineering/planning.md]: I think we will flesh this section out when we get close to doing the retro for Q4

**Comment by jasonyavorsky** [comment]
Done: https://github.com/sourcegraph/handbook/pull/948

**Pull request merged**
